app/controllers/general.php

```diff
@@ -350,13 +351,26 @@ function router(App $utopia, Database $dbForConsole, callable $getProjectDB, Swo
         $utopia->getRoute()?->label('error', '');
         return false;
     } else {
-        throw new AppwriteException(AppwriteException::GENERAL_SERVER_ERROR, 'Unknown resource type ' . $type);
+        throw new AppwriteException(AppwriteException::GENERAL_ACCESS_FORBIDDEN, 'Unknown resource type ' . $type);
     }
 
     $utopia->getRoute()?->label('error', '');
     return false;
 }
 
+App::init()
+    ->groups(['database', 'function', 'storage', 'messaging'])
```

Comment:
Functions is using a plural for group:

```suggestion
    ->groups(['database', 'functions', 'storage', 'messaging'])
```